### PR TITLE
Refactor the logic of getting mentioned accounts and avoid panic there

### PR DESCRIPTION
**Pull request by ono-max**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/executor/waitapproval/waitapproval.go [hunk 1]
```diff
@@ -126,7 +126,7 @@ func (e *Executor) reportRequiringApproval(ctx context.Context) {
 	var approvers []string
 
 	for _, v := range ds.GenericDeploymentConfig.DeploymentNotification.Mentions {
-		if v.Event == "DEPLOYMENT_WAIT_APPROVAL" {
+		if e := "EVENT_" + v.Event; e == model.NotificationEventType_EVENT_DEPLOYMENT_WAIT_APPROVAL.String() {
```
Comment (khanhtc1202) [inline pkg/app/piped/executor/waitapproval/waitapproval.go:129]: `ds.GenericDeploymentConfig.DeploymentNotification` in L128 is nullable.

Comment (ono-max) [reply pkg/app/piped/executor/waitapproval/waitapproval.go]: 5c13de0

File: pkg/app/piped/executor/waitapproval/waitapproval.go [hunk 2]
```diff
@@ -125,9 +125,12 @@ func (e *Executor) reportRequiringApproval(ctx context.Context) {
...
 
-	for _, v := range ds.GenericDeploymentConfig.DeploymentNotification.Mentions {
-		if v.Event == "DEPLOYMENT_WAIT_APPROVAL" {
-			approvers = v.Slack
+	if ds.GenericDeploymentConfig.DeploymentNotification != nil {
+		for _, v := range ds.GenericDeploymentConfig.DeploymentNotification.Mentions {
+			if e := "EVENT_" + v.Event; e == model.NotificationEventType_EVENT_DEPLOYMENT_WAIT_APPROVAL.String() {
+				approvers = v.Slack
+				break
+			}
```
Comment (ono-max) [inline pkg/app/piped/executor/waitapproval/waitapproval.go:133]: Should I have another method to make MentionedAccounts?

Comment (ono-max) [reply pkg/app/piped/executor/waitapproval/waitapproval.go]: e0b6ae4

File: pkg/app/piped/executor/waitapproval/waitapproval.go [hunk 3]
```diff
@@ -117,26 +118,36 @@ func (e *Executor) checkApproval(ctx context.Context) (string, bool) {
...
-			MentionedAccounts: approvers,
+			MentionedAccounts: accounts,
 		},
 	})
 }
+
+func (e *Executor) getMentionedAccounts(ctx context.Context) ([]string, error) {
+	ds, err := e.TargetDSP.GetReadOnly(ctx, e.LogPersister)
+	if err != nil {
+		return nil, fmt.Errorf("failed to prepare running deploy source data (%v)", err)
```
Comment (nakabonne) [inline pkg/app/piped/executor/waitapproval/waitapproval.go:139]: nit
```suggestion
		return nil, fmt.Errorf("failed to prepare running deploy source data: %w", err)
```

**Comment by nakabonne** [comment]
Always amazing how fast you worked
/lgtm

**Comment by khanhtc1202** [comment]
🚀 
/approve

**Pull request merged**